webhooks-extension/src/components/WebhookDisplayTable/WebhookDisplayTable.js

```diff
@@ -134,48 +136,66 @@ export class WebhookDisplayTable extends Component {
       return deleteWithinTimePromise;
     })
 
-    Promise.all(deletePromises).then( () => {
-      this.fetchWebhooksForTable();
-      if(this.state.webhooks.length - rowsToUse.length === 0){
-        this.props.setshowLastWebhookDeletedNotification(true);
-      }
-      else {
+    const deletionError = () => (false);
+    const deletionSuccess = () => (true);
+
+    Promise.all(
+      deletePromises.map(promise =>
+        promise.then(deletionSuccess).catch(deletionError)
```

Comment:
What's the purpose of this? If the array contains false we throw an error and end up in the catch below anyway.

Reply:
So when attempting to delete multiple webhooks and failing, the successfully deleted ones were not being removed. Therefore, I decided to check each individual promise first to see how that deletion went. Those respective promise results were deduced inside `.all()` by the calling of `then()` if successful and returning `true` or catch() if there was an error and returning `false`.  After mapping through each promise in the `.all()` call, we then check if the array returned contains a `false` which denotes that an error occurred in one of the mapped promises. If this happens, the control flow goes to the else statement and throws an Error which will ultimately make it go to the catch callback of the `.all()` method and here is were we change the state to show the error notification with the necessary info. Now, if the array is completely true then move along and check if those were all the webhooks in the table and if so, render the `CreateWebhook` component otherwise just show notification stating that the selected webhooks were deleted. 


Now i did try it without `.then(deletionSuccess).catch(deletionError)` and just having `deletePromises.map(promise => promise)` and although it works, i did notice not being as smooth and in some occasions it didn't even remove the successfully deleted webhooks.